SimDataFormats/Associations/interface/MtdRecoClusterToSimLayerClusterAssociationMap.h

```diff
@@ -0,0 +1,57 @@
+#ifndef SimDataFormats_Associations_MtdRecoClusterToSimLayerClusterAssociationMap_h
+#define SimDataFormats_Associations_MtdRecoClusterToSimLayerClusterAssociationMap_h
+
+#include "DataFormats/Provenance/interface/ProductID.h"
+#include "DataFormats/Common/interface/HandleBase.h"
+#include "DataFormats/FTLRecHit/interface/FTLClusterCollections.h"
+#include "SimDataFormats/CaloAnalysis/interface/MtdSimLayerClusterFwd.h"
+
+#include <vector>
+#include <utility>
+#include <algorithm>
+
+/**
+ * Maps FTLCluserRef to SimLayerClusterRef
+ *
+ */
+class MtdRecoClusterToSimLayerClusterAssociationMap {
+public:
+  using key_type = FTLClusterRef;
+  using mapped_type = MtdSimLayerClusterRef;
+  using value_type = std::pair<key_type, std::vector<mapped_type>>;
+  using map_type = std::vector<value_type>;
+  using const_iterator = typename map_type::const_iterator;
+  using range = std::pair<const_iterator, const_iterator>;
+
+  /// Constructor
+  MtdRecoClusterToSimLayerClusterAssociationMap();
+  /// Destructor
+  ~MtdRecoClusterToSimLayerClusterAssociationMap();
+
+  void emplace_back(const FTLClusterRef& recoClus, std::vector<MtdSimLayerClusterRef>& simClusVect) {
+    map_.emplace_back(recoClus, simClusVect);
+  }
+
+  void sort() { std::sort(map_.begin(), map_.end(), compare); }
```

Comment:
If the function would be called `post_insert()`, the framework would call it automatically when the data product is put in the Event, that would allow to not to rely on the developer to remember to call the `sort()` in the EDProducer.

Reply:
@makortel. Thank you. PR updated to use post_insert()